File: Week_6/problem_3-Print_from_N_to_1.cpp
// https://codeforces.com/group/MWSDmqGsZm/contest/223339/problem/C

#include <iostream>
using namespace std;

void num_reverse_print_recur(int n)
{
    if (n <= 0)
        return;
    cout << n << " ";
    num_reverse_print_recur(n - 1);
}

int main()
{
    int n;
    cin >> n;

    num_reverse_print_recur(n);

    return 0;
}

File: Week_6/problem_3-Print_from_N_to_1.py
# https://codeforces.com/group/MWSDmqGsZm/contest/223339/problem/C

def num_reverse_print_recur(n):
    if n <= 0:
        return
    print(n, end=" ")
    num_reverse_print_recur(n - 1)

n = int(input())
num_reverse_print_recur(n)
